src/Compilers/CSharp/Test/Semantic/Semantics/StackAllocInitializerTests.cs

```diff
@@ -2014,5 +1950,147 @@ static int M(System.Action action)
                 Diagnostic(ErrorCode.ERR_StackallocInCatchFinally, "stackalloc     [ ] { 1, 2, 3 }").WithLocation(96, 34)
                 );
         }
+
+        [Fact]
+        public void StackAllocArrayCreationExpression_Symbols()
+        {
+            var comp = CreateCompilationWithMscorlibAndSpan(@"
+using System;
+unsafe class Test
+{
+    public void Method1()
+    {
+        var obj1 = stackalloc double[2] { 1, 1.2 };
+        Span<double> obj2 = stackalloc double[2] { 1, 1.2 };
```

Comment:
Can we have one more test similar to this with self-referential initializers? 
Like 
`Span<double> obj2 = stackalloc double[2] { obj2[0], obj2.Length };`
`double* obj2 = stackalloc double[2] { obj2[0], *obj2};`

These are error cases, but the semantic model should not go into infinite loop or something like that.

Reply:
@VSadov Added. Also, please review https://github.com/dotnet/roslyn/pull/25384 which contains fixups for merge.